File: MenuAPI/MenuAPI/Models/KhachHang.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace MenuAPI.Models
{
    [Table("khach_hang")]
    public partial class KhachHang
    {
        public KhachHang()
        {
            HoaDons = new HashSet<HoaDon>();
        }

        [Key]
        [Column("khach_hang_id")]
        public int KhachHangId { get; set; }
        [Column("ten_khach_hang")]
        [StringLength(100)]
        public string TenKhachHang { get; set; } = null!;
        [Column("so_dien_thoai")]
        [StringLength(15)]
        public string? SoDienThoai { get; set; }
        [Column("dia_chi")]
        [StringLength(255)]
        public string? DiaChi { get; set; }

        [InverseProperty("KhachHang")]
        public virtual ICollection<HoaDon> HoaDons { get; set; }
    }
}

File: menuapp/lib/Model/khachHang.dart
class KhachHang {
  int khachHangId;
  String tenKhachHang;
  String soDienThoai;
  String diaChi;

  KhachHang({
    required this.khachHangId,
    required this.tenKhachHang,
    required this.soDienThoai,
    required this.diaChi,
  });

  factory KhachHang.fromJson(Map<String, dynamic> json) {
    return KhachHang(
      khachHangId: json['khachHangId'],
      tenKhachHang: json['tenKhachHang'],
      soDienThoai: json['soDienThoai'],
      diaChi: json['diaChi'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'khachHangId': khachHangId,
      'tenKhachHang': tenKhachHang,
      'soDienThoai': soDienThoai,
      'diaChi': diaChi,
    };
  }
}